src/components/Item/VisualizationItem/Visualization/getFilteredVisualization.js

```diff
@@ -0,0 +1,42 @@
+const getFilteredVisualization = (visualization, filters) => {
+    if (!Object.keys(filters).length) {
+        return visualization
+    }
+
+    // deep clone objects in filters to avoid changing the visualization in the Redux store
+    const visRows = visualization.rows.map(obj => ({ ...obj }))
+    const visColumns = visualization.columns.map(obj => ({ ...obj }))
+    const visFilters = visualization.filters.map(obj => ({ ...obj }))
+
+    Object.keys(filters).forEach(dimensionId => {
+        if (filters[dimensionId]) {
+            let dimensionFound = false
```

Comment:
There's really a lot going on here... I'm generally not a fan of `xxFound = true` and the three nested `forEach` loops makes it quite hard to read. I'm sure this could be solved in a simpler way by e.g. checking if the `length` of the matched items is 0 and then push to `visFilters`. But I'd need proper test data to write it and TBH I'm not sure if the readability would actually be increased. 
However if you've got time I'd consider refactoring this and if not at least change the prop name `obj` to something that gives semantic meaning and include a comment with the object structure of the `visualization` props so one knows what to expect.

Reply:
Agreed that this function has a lot going on. I'll add this to the tech debt list.